implementations/rust/ockam/ockam_transport_ble/src/driver/mod.rs

```diff
@@ -0,0 +1,185 @@
+//! Bluetooth platform drivers
+
+#![allow(unused_imports, unused_mut, unused_variables)]
+
+/// support for ST Micro BlueNRG Bluetooth radios
+#[cfg(feature = "use_bluetooth_hci")]
+pub mod bluetooth_hci;
+
+/// support for OS bluetooth
+#[cfg(all(
+    feature = "use_btleplug",
+    any(target_os = "linux", target_os = "macos", target_os = "windows")
+))]
+pub mod btleplug;
+
+#[cfg(not(feature = "std"))]
+mod mutex;
+mod packet;
+mod stream;
+
+pub(crate) use packet::PacketBuffer;
+pub(crate) use stream::{AsyncStream, Sink, Source};
+
+use crate::BleAddr;
+use ockam_core::{async_trait, compat::boxed::Box};
+
+/// The minimum MTU required by the BLE spec. Many devices
+/// (e.g. bluenrg_ms) don't allow for configuration higher than this.
+pub const MTU: usize = 23;
+
+/// Maximum length of characteristic values (250 is the longest
+/// allowed by the BLE specification)
+///
+/// Hard-limited for now according to MTU:
+///
+/// MTU - 5 (packet fields) = 18 bytes of payload
+pub const CHARACTERISTIC_VALUE_LENGTH: usize = MTU - 5;
+
+/// Maximum length of ockam messages
+pub const MAX_OCKAM_MESSAGE_LENGTH: usize = 1024;
+
+/// Ockam BLE service and endpoints are encoded as UUID v5 identifiers
+/// within the ns:OID namespace:
+///
+///     iso.identified-organization.ockam.transport.ble (1.3.214.9.4)
+///
+/// Ockam OID information can be obtained at:
+///
+///     http://oid-info.com/get/1.3.214.9.4
```

Comment:
This link doesn't resolve. Should it?

Reply:
It's not live yet, still bumbling my way through the registration process!